xwiki-platform-core/xwiki-platform-ratings/xwiki-platform-ratings-ui/src/test/java/org/xwiki/ratings/RatingsTest.java

```diff
@@ -0,0 +1,133 @@
+        velocityConfigProps.put(RuntimeConstants.RESOURCE_LOADER + ".class." + RuntimeConstants.RESOURCE_LOADER_CLASS,
+            ClasspathResourceLoader.class.getName());
+        velocityConfigProps.put(RuntimeConstants.VM_LIBRARY, "/templates/macros.vm");
+        velocityConfiguration = this.oldcore.getMocker().registerMockComponent(VelocityConfiguration.class);
+        when(velocityConfiguration.getProperties()).thenReturn(velocityConfigProps);
+
+        // The Ratings macro page includes the rating_macros.vm template.
+        this.oldcore.getMocker().registerMockComponent(TemplateManager.class);
+
+        MacroDescriptor templateMacroDescriptor = mock(MacroDescriptor.class);
+        when(this.templateMacro.getDescriptor()).thenReturn(templateMacroDescriptor);
+        when(templateMacroDescriptor.getParametersBeanClass()).thenReturn((Class) "".getClass());
+
+        registerVelocityTool("escapetool", new EscapeTool());
+
+        // Initialize a document reference to be used as parameter when calling displayFullRating.
+        ModelScriptService modelScriptService = mock(ModelScriptService.class);
+        when(modelScriptService.createDocumentReference("a", "b", "c\"d"))
+            .thenReturn(new DocumentReference("a", "b", "c\"d"));
+        this.oldcore.getMocker().registerComponent(ScriptService.class, "model", modelScriptService);
+
+        // Initialize the ratings script service to allow displayFullRating to display the blocks to be tested.
+        RatingsScriptService ratingsScriptService = mock(RatingsScriptService.class);
+        when(ratingsScriptService.getAverageRating(any())).thenReturn(Optional.of(mock(AverageRating.class)));
+        this.oldcore.getMocker().registerComponent(ScriptService.class, "ratings", ratingsScriptService);
+    }
+
+    @Test
+    void displayFullRatingEscapesRequestDocRef() throws Exception
+    {
+        setOutputSyntax(Syntax.XHTML_1_0);
+
+        // Load the RatingsMacros that contains the tested displayFullRating.
+        loadPage(new DocumentReference("xwiki", Arrays.asList("XWiki", "Ratings"), "RatingsMacros"));
+
+        // Call the displayFullRating macro with a name with a double quote to test that it is properly escaped.
+        String script = "{{include reference=\"XWiki.Ratings.RatingsMacros\"/}}\n"
+            + "{{velocity}}#set($tdoc = $services.model.createDocumentReference('a', 'b', 'c\"d'))\n"
+            + "#displayFullRating($tdoc){{/velocity}}";
+
+        XWikiDocument document = new XWikiDocument(new DocumentReference("xwiki", "XWiki", "Test"));
+        document.setSyntax(Syntax.XWIKI_2_0);
+        document.setContent(script);
+        this.xwiki.saveDocument(document, "registering document", true, this.context);
+        String content = document.getRenderedContent(this.context);
+
+        // Verify that the displayFullRating parameter is actually escaped when used.
+        assertTrue(content.contains("data-reference=\"a:b.c&#34;d\""));
```

Comment:
Maybe something more precise in term of HTML asserting? Some code could use a `data-reference` attribute not related to ratings for ex, so doing a check more related to ratings could make the test more stable. WDYT?

Reply:
I've replaced this test with an inspection of the DOM structure with Jsoup.